src/com/facebook/buck/swift/SwiftCompile.java

```diff
@@ -65,13 +66,14 @@
 import java.io.IOException;
 import java.nio.file.Files;
 import java.nio.file.Path;
+import java.nio.file.Paths;
 import java.util.Arrays;
 import java.util.Optional;
 import java.util.SortedSet;
 import java.util.function.Function;
 
 /** A build rule which compiles one or more Swift sources into a Swift module. */
-public class SwiftCompile extends AbstractBuildRule {
+public class SwiftCompile extends AbstractBuildRule implements SupportsInputBasedRuleKey {
```

Comment:
This change can potentially break things because there's still a bunch of fields which are not part of the rulekey. This becomes problematic if any of those unannotated keys contains data computed by some transitive dependency and that data is part of the transitive rule's rulekey.

I'd need to audit all of the remaining fields to ensure that this is the case but I haven't had the time to do that yet.

On the correctness note, a quick look reveals that some stuff is already broken: e.g., `enableObjcInterop` should have an `@AddToRuleKey` annotation.

Reply:
I see, thx for the review.

Would it be enough for us to review all the "private final" fields in `SwiftCompile` class? Or do we need to review the fields in its super class as well?

If former, then the fields we shall review would be the following:
```
  private final Path objectFilePath;
  private final Path modulePath;
  private final Path moduleObjectPath;
  private final ImmutableList<Path> objectPaths;
  private final Optional<AbsPath> swiftFileListPath;
  private final Path headerPath;
  private final CxxPlatform cxxPlatform;
  private final ImmutableSet<FrameworkPath> frameworks;
  private final boolean enableObjcInterop;
  private final SwiftBuckConfig swiftBuckConfig;
  private BuildableSupport.DepsSupplier depsSupplier;
```

Which means in order to safely apply this change, **we should review these fields and make sure all of them whenever necessary should have that `@AddToRuleKey` annotation.**

Is my understanding above correct? Thanks!